scripts/prepare_heat_demand.py

```diff
@@ -2,15 +2,487 @@
+
+    Parameters
+    ----------
+    heat_load_postprocessed : pd.Dataframe
+        Empty Dataframe as result DataFrame
+    heat_load_year : pd.Dataframe
+        DataFrame with total normalized heat load in year to be processed
+    region : str
+        Region (eg. Brandenburg)
+    scenario : str
+        Scenario e.g. "base
+    unit : str
+        Unit of total demands (eg. GWh)
+
+    Returns
+    -------
+    heat_load_postprocessed : pd.DataFrame
+         DataFrame that contains stacked heat load profile
+
+    """
+    # Stack time series with total heat load in year
+    heat_load_year_stacked = dp.stack_timeseries(heat_load_year)
+
+    heat_load_year_stacked["region"] = region
+    heat_load_year_stacked["scenario"] = scenario
+    heat_load_year_stacked["var_unit"] = unit[0]
+
+    # Append stacked heat load of year to stacked time series with total heat loads
+    heat_load_postprocessed = pd.concat(
+        [heat_load_postprocessed, heat_load_year_stacked], ignore_index=True, sort=False
+    )
+
+    return heat_load_postprocessed
+
+
+if __name__ == "__main__":
+    in_path1 = sys.argv[1]  # path to weather data
+    in_path2 = sys.argv[2]  # path to household distributions data
+    in_path3 = sys.argv[3]  # path to holidays
+    in_path4 = sys.argv[4]  # path to building class
+    in_path5 = sys.argv[5]  # path to b3 scalars.csv
+    out_path = sys.argv[6]
+
+    REGION = ["BB", "B"]
+    SCENARIO = "base"
+    CARRIERS = ["heat_central", "heat_decentral"]
+
+    # Add empty data frame for results / output
+    total_heat_load = pd.DataFrame(columns=dp.HEADER_B3_TS)
+
+    # Read state heat demands of ghd and hh sectors
+    sc = dp.load_b3_scalars(in_path5)
+
+    for region in REGION:
+        share_efh, share_mfh = get_shares_from_hh_distribution(in_path2, region)
+
+        weather_file_names = find_regional_files(in_path1, region)
+
+        for weather_file_name, carrier in itertools.product(
+            weather_file_names, CARRIERS
+        ):
+            # Read year from weather file name
+            year = get_year(weather_file_name)
+
+            # Get holidays
+            holidays = get_holidays(year, region, in_path3)
+
+            # Read temperature from weather data
+            path_weather_data = os.path.join(in_path1, weather_file_name)
+            temperature = pd.read_csv(path_weather_data, usecols=["temp_air"], header=0)
+
+            # Get building class
+            building_class = get_building_class(region, in_path4)
+
+            # Get heat demand in region and scenario
+            yearly_demands, sc_demand_unit = get_heat_demand(
+                sc, SCENARIO, carrier, region
+            )
+
+            heat_load_year = calculate_heat_load(
+                carrier, holidays, temperature, yearly_demands, building_class
+            )
+            total_heat_load = postprocess_data(
+                total_heat_load, heat_load_year, region, f"ts_{year}", sc_demand_unit
+            )
+
```

Comment:
Can you add a function here that sums up the scalars for hh, ghd and i?

Reply:
Done with commit https://github.com/rl-institut/oemof-B3/pull/76/commits/a8a7b1f4a6f9639c1f1c174ec0fb79a8804d4cbe and https://github.com/rl-institut/oemof-B3/pull/76/commits/a8a7b1f4a6f9639c1f1c174ec0fb79a8804d4cbe with potential for more efficient and cleaner coding. Also, I don't know how to get rid of the `SettingWithCopyWarning` which occurs when I intentionally overwrite values in a DataFrame.